File: programmers/level2/프린터.js
function solution(priorities, location) {
    var answer = 0;
    
    var new_priorities = priorities
        .slice()
        .sort((a, b) => {
            if(a > b) return -1;
            else return 1;
        })
        .filter(priority => priority >= priorities[location]);
    
    var foundAnswer = false;
    var location_cur = -1;
    const priorities_length = priorities.length;
    for(var i = 0; !foundAnswer && i < new_priorities.length; i++) {
        var foundPrint = false;
        for(var j = 0; !foundPrint && j < priorities_length; j++) {
            location_cur++;
            if(location_cur >= priorities_length) location_cur -= priorities_length;
            
            if(priorities[location_cur] === new_priorities[i]) {
                answer++;
                foundPrint = true;
                if(location_cur === location) foundAnswer = true;
            }
        }
    }
    
    return answer;
}

File: programmers/스택ㅣ큐/프린터.py
from typing import List


def solution(priorities: List[int], location: int) -> int:
    answer: int = 0
    prev: int = -1

    for priority in sorted(set(priorities), reverse=True):
        if priority < priorities[location]:
            break

        same_priorities = [idx for idx, pri in enumerate(priorities) if priority == pri]
        print_orders = list(filter(lambda x: x > prev, same_priorities)) + list(
            filter(lambda x: x < prev, same_priorities)
        )
        if location in print_orders:
            answer += print_orders.index(location) + 1
        else:
            answer += len(print_orders)

        prev = print_orders[-1]

    return answer


# 1
print(solution([2, 1, 3, 2], 2))
# 5
print(solution([1, 1, 9, 1, 1, 1], 0))
